File: 643-maximum-average-subarray-i/maximum-average-subarray-i.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
           //sliding window approach

           //precalculating window element;
           int n=nums.size();
            double sum=0;
           for(int i=0;i<k;i++){
            sum+=nums[i];
           }   
           double maxsum=sum;
           for(int j=k;j<nums.size();j++){
            sum+=nums[j]-nums[j-k];
            maxsum=max(sum,maxsum);
           }
           return maxsum/k;
    }
};

File: 643-maximum-average-subarray-i/maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        //pre Calculating the value first
        double sum=0;
        for(int i=0;i<k;i++){
            sum+=nums[i];
        }
        double maxsum=sum;
        for(int j=k;j<nums.length;j++){
            sum-=nums[j-k];
            sum+=nums[j];
            maxsum=Math.max(sum,maxsum);
        }
        return (double)maxsum/k;
    }
}